src/main/steps/common/keep-details-private/KeepYourDetailsPrivateMapper.ts

```diff
@@ -1,65 +1,76 @@
+  const { startAlternative, contactDetailsPrivate, detailsKnown } = req.session.userCase;
+  const request: Partial<PartyDetails> = {};
+
   const confidentialityList: ConfidentialityList[] = [];
 
-  if (req.session.userCase.startAlternative === YesOrNo.YES && req?.session?.userCase?.contactDetailsPrivate) {
-    req.session.userCase.contactDetailsPrivate.forEach(element => {
+  if (startAlternative === YesOrNo.YES && contactDetailsPrivate) {
+    contactDetailsPrivate.forEach(element => {
       confidentialityList.push(ConfidentialityList[element]);
     });
   }
 
-  if (partyDetails.response && partyDetails.response?.keepDetailsPrivate) {
-    keepDetailsPrivate = partyDetails.response?.keepDetailsPrivate;
-    keepDetailsPrivate.otherPeopleKnowYourContactDetails = req.session.userCase.detailsKnown!;
-    keepDetailsPrivate.confidentiality = req.session.userCase.startAlternative!;
-    keepDetailsPrivate.confidentialityList = confidentialityList;
-  } else {
-    partyDetails.response = {
+  Object.assign(request, {
+    response: {
       keepDetailsPrivate: {
-        otherPeopleKnowYourContactDetails: req.session.userCase.detailsKnown!,
-        confidentiality: req.session.userCase.startAlternative!,
+        otherPeopleKnowYourContactDetails: detailsKnown,
+        confidentiality: startAlternative,
         confidentialityList,
       },
-    };
+    },
+  });
+
+  if (startAlternative === YesOrNo.NO) {
+    delete request.response?.keepDetailsPrivate?.confidentialityList;
   }
-  return partyDetails;
+
+  return request;
 };
-export const getKeepYourDetailsPrivate = (partyDetails: PartyDetails, req: AppRequest): Partial<CaseWithId> => {
-  req.session.userCase.detailsKnown = partyDetails.response.keepDetailsPrivate?.otherPeopleKnowYourContactDetails;
-  req.session.userCase.startAlternative = partyDetails.response.keepDetailsPrivate?.confidentiality;
-  const confidentialityList: string[] = [];
-  if (
-    partyDetails.response.keepDetailsPrivate?.confidentiality === YesOrNo.YES &&
-    partyDetails.response.keepDetailsPrivate.confidentialityList
-  ) {
-    partyDetails.response.keepDetailsPrivate.confidentialityList.forEach(element => {
-      confidentialityList.push(ConfidentialityList[element]);
+
+export const mapKeepYourDetailsPrivate = (partyDetails: PartyDetails): Partial<CaseWithId> => {
+  const privateDetails = {};
+  const { keepDetailsPrivate } = partyDetails.response ?? {};
+
+  let confidentialityList: string[] | undefined = [];
+  if (keepDetailsPrivate?.confidentiality === YesOrNo.YES && keepDetailsPrivate.confidentialityList) {
+    keepDetailsPrivate.confidentialityList.forEach(element => {
+      confidentialityList?.push(ConfidentialityList[element]);
     });
-    req.session.userCase.contactDetailsPrivate = confidentialityList;
   }
 
-  return req.session.userCase;
+  if (confidentialityList.length === 0) {
+    confidentialityList = undefined;
```

Comment:
why is confidentialityList assigned to "undefined" when the type of confidentialityList is array in definition.ts.